### PR TITLE
add selectionSetList to pipeline function

### PR DESCRIPTION
**Pull request by kldeb**
*Issue #, if available:*
I didn't open an issue for this, but chatted on discord about it

*Description of changes:*
This change adds the selectionSetList from the ctx object to the payload so that it can be used in the lambda function. We could use this on our project so I modified the resolver my selection. I'm opening a PR because I think it's important enough a feature with no side effects that I can think of.

Think of the case where you need to make multiple calls to third party REST APIs to retrieve child data etc and you would like to do it async on the server rather than the client for two reasons I can think of: 

1. simplifies the front end for front end devs
2. reduce the number of round trips. 

The lambda resolver can now inspect in the selection set list and only make additional calls if the field is requested from the client. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by SwaySway — COMMENTED** [review]
@kldeb 
Changes look good are you able to provide a unit/snapshot test checking the resolver output?

**Comment by kldeb** [comment]
@SwaySway sure, can you point me to a good example to imitate?

File: packages/graphql-function-transformer/src/FunctionTransformer.ts [hunk 1]
```diff
@@ -148,14 +148,15 @@ export class FunctionTransformer extends Transformer {
             source: ref('util.toJson($ctx.source)'),
             request: ref('util.toJson($ctx.request)'),
             prev: ref('util.toJson($ctx.prev)'),
+            selectionSetList: str('$ctx.info.selectionSetList'),
```
Comment (SwaySway) [inline packages/graphql-function-transformer/src/FunctionTransformer.ts:151]: Can you make this a ref as well?

**Review by SwaySway — COMMENTED** [review]
@kldeb 
Apologies for the late response, are you able to add the changes to this PR?

If so here is the `@http` unit tests for checking resolver output. 
https://github.com/aws-amplify/amplify-cli/blob/master/packages/graphql-http-transformer/src/__tests__/HttpTransformer.test.ts

**Pull request closed without merging**